### DISCUSSION TITLE
Utilizing identities in a CRUD app

### DISCUSSION THREAD

**Discussion by nedpals** [ory/kratos#1245] [Q&A] [answered]
Hi! First of all, thanks for making this. Second, apologies if I sound newbie-ish. 

I'm building a web application using Go and I'm trying to see if Kratos would fit my needs. Skimming through the documentation and through other resources didn't satisfied my curiosity. My question is:

1. How would I reference a user/identity to a data in a CRUD app? Would the identity ID be enough?

2. Is there a public API/function in the SDK that I can use to retrieve a specific user similar to Auth0?

Thanks in advance :)

**Comment by Benehiko** [comment, marked as answer]
Hi @nedpals,

Thanks for reaching out :)

Yes you use the user ID for that. This can easily be done in your application using a middleware to retrieve the user ID from kratos using its cookie https://www.ory.sh/kratos/docs/reference/api#check-who-the-current-http-session-belongs-to

Retrieving an Identity can also be done using its ID https://www.ory.sh/kratos/docs/reference/api#get-an-identity

**Reply by nedpals** [reply]
> Retrieving an Identity can also be done using its ID https://www.ory.sh/kratos/docs/reference/api#get-an-identity

Since this is an admin API, does that mean that I have to fetch and control the data to be sent first before exposing it into the public?

**Reply by Benehiko** [reply]
So Admin APIs are for your server-side application to set or get sensitive information from Kratos. If you want to get the identity client side, just request the public endpoint https://www.ory.sh/kratos/docs/reference/api#check-who-the-current-http-session-belongs-to and it will respond with 200 for authenticated with a `session` payload containing the user ID and traits. This endpoint uses the cookie issued to the client on login.

**Reply by nedpals** [reply]
Ah, I get it now. I must've missed that part. Thanks!
